diff --git a/dist/declarations/src/three-types.d.ts b/dist/declarations/src/three-types.d.ts
index 7668a42e27ac2c51ebaaac88db8bfd8fe823ae75..454b496ddf130ff99c462e3998553ce11a4d654e 100644
--- a/dist/declarations/src/three-types.d.ts
+++ b/dist/declarations/src/three-types.d.ts
@@ -386,8 +386,10 @@ export interface ThreeElements {
 }
 declare global {
     namespace JSX {
-        interface IntrinsicElements extends ThreeElements {
+        interface IntrinsicElements extends Pick<ThreeElements, 'mesh' | 'meshBasicMaterial' | 'sprite' | 'spriteMaterial'> {
         }
     }
+    interface JSX_IntrinsicElements extends ThreeElements {
+    }
 }
 export {};
