diff --git a/dist/declarations/src/three-types.d.ts b/dist/declarations/src/three-types.d.ts
index 0743aa43ca1b12328233b32c10c31b4f2376efd5..a5abf4a91036260a0b9ec4e1d8f51fd47f332450 100644
--- a/dist/declarations/src/three-types.d.ts
+++ b/dist/declarations/src/three-types.d.ts
@@ -391,7 +391,7 @@ export interface ThreeElements {
 }
 declare global {
     namespace JSX {
-        interface IntrinsicElements extends ThreeElements {
+        interface JSX_IntrinsicElements extends ThreeElements {
         }
     }
 }
