diff --git a/dist/declarations/src/three-types.d.ts b/dist/declarations/src/three-types.d.ts
index 7f809d578d388ca8b1031efe452c4eba7ba5d135..a5c7ee7625361a64953395d209f7a02135c5b3fb 100644
--- a/dist/declarations/src/three-types.d.ts
+++ b/dist/declarations/src/three-types.d.ts
@@ -386,25 +386,29 @@ export interface ThreeElements {
 }
 declare global {
     namespace JSX {
-        interface IntrinsicElements extends ThreeElements {
+        interface IntrinsicElements extends Pick<ThreeElements, 'mesh' | 'meshBasicMaterial' | 'sprite' | 'spriteMaterial'> {}
+        interface JSX_IntrinsicElements extends ThreeElements {
         }
     }
 }
 declare module 'react' {
     namespace JSX {
-        interface IntrinsicElements extends ThreeElements {
+        interface IntrinsicElements extends Pick<ThreeElements, 'mesh' | 'meshBasicMaterial' | 'sprite' | 'spriteMaterial'> {}
+        interface JSX_IntrinsicElements extends ThreeElements {
         }
     }
 }
 declare module 'react/jsx-runtime' {
     namespace JSX {
-        interface IntrinsicElements extends ThreeElements {
+        interface IntrinsicElements extends Pick<ThreeElements, 'mesh' | 'meshBasicMaterial' | 'sprite' | 'spriteMaterial'> {}
+        interface JSX_IntrinsicElements extends ThreeElements {
         }
     }
 }
 declare module 'react/jsx-dev-runtime' {
     namespace JSX {
-        interface IntrinsicElements extends ThreeElements {
+        interface IntrinsicElements extends Pick<ThreeElements, 'mesh' | 'meshBasicMaterial' | 'sprite' | 'spriteMaterial'> {}
+        interface JSX_IntrinsicElements extends ThreeElements {
         }
     }
 }
